File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/abstract_factory/solution/conventional_garden.cpp
// conventional_garden.cpp

#include "conventional_garden.h"
#include "conventional_lettuce.h"
#include "conventional_tomato.h"
#include "garden.h"

conventional_garden::conventional_garden(conventional_lettuce * a_lettuce_prototype)
: garden(a_lettuce_prototype)
{
}

void conventional_garden::set_lettuce_prototype(conventional_lettuce * a_lettuce_prototype)
{
	garden::set_lettuce_prototype(a_lettuce_prototype);
}

conventional_tomato * conventional_garden::make_tomato() const
{
	return new conventional_tomato(1, 2);
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/abstract_factory/solution/ConventionalGarden.m
% ConventionalGarden.m

classdef ConventionalGarden < Garden
	methods
		% constructor:
		function theConventionalGarden = ConventionalGarden(aLettucePrototype)
			if ~isa(aLettucePrototype, 'ConventionalLettuce')
				error('An ConventionalGarden must have a ConventionalLettuce prototype.');
			end
			theConventionalGarden@Garden(aLettucePrototype);
		end

		function result = makeTomato(theGarden)
			result = ConventionalTomato(1, 2);
		end
	end
end
